variables.tf

```
@@ -1,9 +1,60 @@
-  type        = string
+variable "enable" {
+  description = "(Optional) Enable GuardDuty monitoring and feedback reporting. Setting to false is equivalent to 'suspending'GuardDuty. Defaults to true."
+  type        = bool
+  default     = true
 }
 
-variable "guardduty_master_detector_id" {
-  description = "GuardDuty Detector ID for master account"
-  type        = string
+variable "filter" {
+  description = "GuardDuty filter configuration"
+  type = list(object({
+    name        = string                   # (Required) The name of your filter.  SPACES ARE NOT ALOWED
+    description = string                   # (Optional) Description of the filter.
+    rank        = number                   # (Required) Specifies the position of the filter in the list of current filters. Also specifies the order in which this filter is applied to the findings.
+    action      = string                   # (Required) Specifies the action that is to be applied to the findings that match the filter. Can be one of ARCHIVE or NOOP.
+    tags        = map(string)              # (Optional) - The tags that you want to add to the Filter resource. A tag consists of a key and a value.
+    criterion = list(object({              # (Represents the criteria to be used in the filter for querying findings. Contains one or more criterion blocks
+      field                 = string       # (Required) The name of the field to be evaluated. The full list of field names can be found in AWS documentation.
+      equals                = list(string) # (Optional) List of string values to be evaluated.
+      not_equals            = list(string) # (Optional) List of string values to be evaluated.
+      greater_than          = string       # (Optional) A value to be evaluated. Accepts either an integer or a date in RFC 3339 format.
+      greater_than_or_equal = string       # (Optional) A value to be evaluated. Accepts either an integer or a date in RFC 3339 format.
+      less_than             = string       # (Optional) A value to be evaluated. Accepts either an integer or a date in RFC 3339 format.
+      less_than_or_equal    = string       # (Optional) A value to be evaluated. Accepts either an integer or a date in RFC 3339 format.
+    }))
+  }))
+  default = []
+}
+
+variable "ipset" {
+  description = "GuardDuty ipset"
+  type = list(object({
+    activate = bool        # (Required) Specifies whether GuardDuty is to start using the uploaded IPSet.
+    format   = string      # (Required) The format of the file that contains the IPSet. Valid values: TXT | STIX | OTX_CSV | ALIEN_VAULT | PROOF_POINT | FIRE_EYE
+    location = string      # (Required) The URI of the file that contains the IPSet.
+    name     = string      # (Required) The friendly name to identify the IPSet.
+    tags     = map(string) # (Optional) Key-value map of resource tags.
+  }))
+  default = []
+}
+
+variable "threatintelset" {
```

Since this is a list, make it plural
```suggestion
variable "threatintelsets" {
```

---

Updated